runelite-client/src/main/java/net/runelite/client/Notifier.java

```diff
@@ -159,18 +189,40 @@ private void sendMacNotification(
 		sendCommand(commands);
 	}
 
-	private void sendCommand(final List<String> commands)
+	private String sendCommand(final List<String> commands)
 	{
 		try
 		{
-			new ProcessBuilder(commands.toArray(new String[commands.size()]))
+			Process proc = new ProcessBuilder(commands.toArray(new String[commands.size()]))
 				.redirectErrorStream(true)
 				.start();
+
+			try (InputStream is = proc.getInputStream())
+			{
+				return CharStreams.toString(new InputStreamReader(is, Charsets.UTF_8));
```

Comment:
Are we sure this doesn't do anything bad? I would think it would cause it to hang until the process exits?

Reply:
Would it make sense to either

1. Keep the previous implementation and make a similar method that actually captures output?
2. Keep the previous implementation and submit a callable that does `which notify-send` to an executor service so no blocking happens until it absolutely needs to?

Reply:
I would maybe run the entire notification on a thread from the executor service. Maybe you could use the exit code from notify-send to judge whether to fall back to tray notify as well (does it exit 1 if it doesn't exist?)

Reply:
Looks like that'll work. Thanks for the suggestions.

```sh
$ which notify-send
/usr/bin/notify-send
$ echo $?
0

$ which nope
$ echo $?
1

Reply:
I mean, don't even use which at all. Just check notify-send failed due to either not existing or some other resaon.

Reply:
Understood, new code should properly do this.

Reply:
It'll exit 127 if it doesn't exist, but given the nature of error codes, if notify-send exits with anything other than 0 when given valid parameters it's safe to assume the notification didn't go through.
That said, did some testing, and: Even if it returns 0, it's not completely safe to say the notification *did* go through, because the return value at https://github.com/GNOME/libnotify/blob/master/tools/notify-send.c#L278 is... totally ignored.
This is problematic if, say, the user has notify-send, but no notification daemon.
So maybe add user configuration for this 'just in case'. In the meantime I'll poke libnotify...

Reply:
Hmm. I would open an issue with libnotify about that and see if it is intentional.

Reply:
So while I *did* poke libnotify, there's been no acknowledgement, not even a bug status change. Not sure what to do here.